sys/kern/uio.c

```diff
@@ -86,3 +88,33 @@ int uiomove_frombuf(void *buf, size_t buflen, struct uio *uio) {
 
   return uiomove((char *)buf + offset, buflen - offset, uio);
 }
+
+int uio_init_from_user_iovec(uio_t *uio, uio_op_t op, const struct iovec *u_iov,
+                             int iovcnt) {
+  int error;
+  if (iovcnt <= 0 || iovcnt > IOV_MAX)
+    return EINVAL;
+  const size_t iov_size = sizeof(iovec_t) * iovcnt;
+  iovec_t *k_iov = kmalloc(M_TEMP, iov_size, 0);
+  if ((error = copyin(u_iov, k_iov, iov_size)))
+    goto err_free;
+  size_t len = 0;
```

Comment:
Maybe `int iovec_length(iovec_t *iov, int iovcnt, size_t *lengthp)` ? Should return EINVAL on error, 0 otherwise.

Reply:
Done.